### PR TITLE
Support Retries

### PR DESCRIPTION
**Pull request by EJH2**
This PR adds support for retries as noted in #54.

- Retries will now fill the bar with red with a notice for how long until the next retry, and then the bar will start over with next progress update
- Created custom retry handling method in JS to allow users to extend current behavior

### CODE REVIEW

**Comment by EJH2** [comment]
Once this gets reviewed I'll see about how to implement the other two states, as well as taking a crack at some other long-standing issues. Just need to get this into the working tree so I can start.

File: README.md [hunk 1]
```diff
@@ -160,6 +160,7 @@ The `initProgressBar` function takes an optional object of options. The followin
 | onProgress | function to call when progress is updated | CeleryProgressBar.onProgressDefault |
 | onSuccess | function to call when progress successfully completes | CeleryProgressBar.onSuccessDefault |
 | onError | function to call on a known error with no specified handler | CeleryProgressBar.onErrorDefault |
+| onRetry | function to call when task a task attempts to retry | CeleryProgressBar.onRetryDefault |
```
Comment (OmarWKH) [inline README.md:163]: Typo: task a task

File: celery_progress/static/celery_progress/celery_progress.js [hunk 2]
```diff
@@ -79,7 +86,13 @@ class CeleryProgressBar {
             if (data.success === true) {
                 this.onSuccess(this.progressBarElement, this.progressBarMessageElement, this.getMessageDetails(data.result));
             } else if (data.success === false) {
-                this.onTaskError(this.progressBarElement, this.progressBarMessageElement, this.getMessageDetails(data.result));
+                if (data.state === 'RETRY') {
+                    this.onRetry(this.progressBarElement, this.progressBarMessageElement, data.result.message, data.result.when);
+                    done = false;
+                    delete data.result;
```
Comment (OmarWKH) [inline celery_progress/static/celery_progress/celery_progress.js:92]: I understand why this is needed. The existing code is confusing and should probably be changed in the future.

**Review by OmarWKH — APPROVED** [review]
Nice work.

In addition to my comment above, we could do the following later:

- Use data.state more often.
- Stop polling until the retry starts.

File: celery_progress/static/celery_progress/celery_progress.js (hunk 2 above)
Comment (EJH2) [reply celery_progress/static/celery_progress/celery_progress.js:92]: I agree, while the previous code works with few states, it does not make extending easy. I originally intended to add more than just RETRY support in this PR, but trying to support REVOKED and IGNORED in this state was a bit hairy.

File: README.md (hunk 1 above)
Comment (EJH2) [reply README.md:163]: Whoops, good catch

**Comment by EJH2** [comment]
If there are no other objections, I'll go ahead merge this in

**Comment by czue** [comment]
Sorry I was slow but this looks great, thanks!

**Pull request merged**